### PR TITLE
Allow user to supply processes to watch on windows.

### PR DESCRIPTION
**Pull request by captncraig**
Previously, only linux processes were able to be added via config file. Windows users only got the hard coded list.

This change allows windows users of scollector to add a regex to the conf file for windows processes and services to monitor.

One thing to consider is that on linux, the `process` line should be of format: `command,name,regex` while on windows, it just expects any regex that matches the process names you want. 

This is a bit difficult to unify across os's, but this is my attempt.

### CODE REVIEW

**Comment by captncraig** [comment]
#575

File: cmd/scollector/collectors/processes_linux.go [hunk 1]
```diff
@@ -1,18 +1,29 @@
 package collectors
 
 import (
+	"bosun.org/metadata"
+	"bosun.org/opentsdb"
```
Comment (kylebrandt) [inline cmd/scollector/collectors/processes_linux.go]: Shouldn't be moved to top, these are always after a space after the normal packages. Using goimports?

Comment (captncraig) [inline cmd/scollector/collectors/processes_linux.go]: My editor fmts, but apparently not in _linux files. 

I checked build in my linux vm, but guess I should fmt too.

**Comment by gbrayut** [comment]
PR looks good to me. I built and ran on windows and linux and both worked using the conf file as expected. Windows probably should use a bit more strict regexes to prevent monitoring the wrong processes since it doesn't have argument match like linux. Also it is matching against the output of `gwmi Win32_PerfRawData_PerfProc_Process | ft name` which does strange things like `powershell` then 'powershell#1,powershell#2,...` for multiple instances with the same name. In fact, we previously were only monitoring the First powershell instance (Doh!)

Proposed baseline conf file for Windows (I'll put this in our sre/scollector-build repo for tracking)

> process=^chrome
> process=^powershell
> process=^scollector
> process=^WinRM
> process=^MSSQLServer
> process=^WmiPrvSE
> process=^java
> process=^snmp
> process=^mscorsvw
> process=^SocketServer
> process=^ChatServer
> process=^StackServer
> process=^LogStasher
> process=^TCBuildAgent
> process=^dsm_(sa|om)

We can then add one off additions to individual servers (like vpxd on vsphere)

Example of graphs on devbosun http://goo.gl/dAXPGw

File: cmd/scollector/main.go [hunk 2]
```diff
@@ -193,11 +189,9 @@ func main() {
 			collectors.Vsphere(user, pwd, host)
 		}
 	}
-	if len(procs) > 0 {
-		if err := collectors.WatchProcesses(procs); err != nil {
-			log.Fatal(err)
-		}
-	}
+	// Start all process collectors. This is platform specific.
```
Comment (kylebrandt) [inline cmd/scollector/main.go]: They don't start here they are just registered. `collectors.Run(c)` starts them.

**Comment by gbrayut** [comment]
LGTM! Confirmed still works on linux and windows. Also confirmed failure when bad regex in conf:

2015/02/25 13:06:59 fatal: main.go:117: error parsing regexp: missing closing ]: `[{)`

**Pull request merged**